benchmark/fs/bench-fchmodSync.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+const common = require('../common');
+const fs = require('fs');
+const tmpdir = require('../../test/common/tmpdir');
+tmpdir.refresh();
+
+const bench = common.createBenchmark(main, {
+  type: ['existing', 'non-existing'],
+  n: [1e3],
+});
+
+function main({ n, type }) {
+  let files;
+
+  switch (type) {
+    case 'existing':
+      files = [];
+
+      // Populate tmpdir with mock files
+      for (let i = 0; i < n; i++) {
+        const path = tmpdir.resolve(`fchmodsync-bench-file-${i}`);
+        fs.writeFileSync(path, 'bench');
+        files.push(path);
+      }
+      break;
+    case 'non-existing':
+      files = new Array(n).fill(tmpdir.resolve(`.non-existing-file-${Date.now()}`));
+      break;
+    default:
+      new Error('Invalid type');
+  }
+
+  const fds = files.map((x) => {
+    // Try to open, if not return invalid fd (-1)
+    try {
+      return fs.openSync(x, 'r');
+    } catch {
+      return -1;
```

Comment:
We validate for Uint32 for the file descriptors. I think this benchmark does not hit the C++ path and just terminates on the JS side.

Reply:
9999? 😅

@bnoordhuis Any suggestions?

Reply:
Check out [`runWithInvalidFD()` from `test/common/index.js`](https://github.com/nodejs/node/blob/bd614f3ce7c893ba32d8c5e0b37747d2fc4132c3/test/common/index.js#L800)

Reply:
Changed to use `1 << 30` like that, also updated desc with new benchmark results.